drgn_tools/scsidevs.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2023, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/
+"""
+Helper to view scsi devices
+"""
+from typing import Iterator
+
+from drgn import cast
+from drgn import FaultError
+from drgn import Object
+from drgn import Program
+from drgn.helpers.common.format import escape_ascii_string
+from drgn.helpers.linux.block import for_each_disk
+from drgn.helpers.linux.block import part_devt
+from drgn.helpers.linux.block import part_name
+from drgn.helpers.linux.device import MAJOR
+from drgn.helpers.linux.device import MINOR
+from drgn.helpers.linux.xarray import xa_for_each
+
+
+def is_blkdev_hdpart(prog: Program) -> bool:
+    """
+    Checks whether partitions are represented by struct block_device
+    or struct hd_struct
+    """
+    has_bdev_struct = True
+    try:
+        for disk in for_each_disk(prog):
+            for bdev in for_each_partition_in_table(disk.part_tbl):
+                return has_bdev_struct
+    except AttributeError:
+        has_bdev_struct = False
+
+    return has_bdev_struct
+
+
+def for_each_partition_in_table(xarray: Object) -> Iterator[Object]:
+    """
+    Helper to iterate through partition table
+    """
+    for _, entry in xa_for_each(xarray.address_of_()):
+        part_ptr = cast("struct block_device *", entry)
+        if not part_ptr.value_():
+            continue
+        yield part_ptr
+
+
+def print_scsi_devs_info(prog: Program) -> None:
+    """
+    Prints the scsi device information
+    """
```

Comment:
So, I'm trying to understand what is scsi-specific here. As far as I can tell, this is simply printing block device info?

If that's the case, then it really doesn't belong in a module named with `scsi`. Probably the `drgn_tools.block` module, and it should be named something like `print_block_devs_info()`.